File: Models/Task.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace TaskListCapstone.Models
{
    public class Task
    {
        [Key]
        //Primary key: unique identifier on each row of our database
        public int Id { get; set; }
        public string TaskName { get; set; }
        public string TaskDescription { get; set; }
        public string AssignedTo { get; set; }
        public DateTime DueDate { get; set; }
        public bool IsCompleted { get; set; }
    }
}

File: ClientApp/src/app/Task.ts
export interface Task{
    id : number;
    taskName : string;
    taskDescription : string;
    assignedTo : string;
    dueDate: Date;    
    isCompleted : boolean;
}

export class Convert {
    public static toTask(json: string): Task {
        return JSON.parse(json);
    }

    public static TaskToJson(value: Task): string {
        return JSON.stringify(value);
    }
}
